### PR TITLE
Add Circular Pathed Platforms

### PR DESCRIPTION
**Pull request by weluvgoatz**
This adds circular-pathed platforms to the game.

Yes, they do follow the same behavior as flames, moving around in a circle.  You can set the radius and speed of their movement as well as a delay for when they begin moving.  You can also change the sprite.

They could possibly be introduced in the ice bridge levels in Icy Island because 1) Those levels introduce no new mechanics and 2) They could "foreshadow" the introduction of the flames by introducing a docile/helpful variant and then showing you the enemy version of it.

This was quite simple of an object to add.  Please test and review, and hopefully this will get merged once it looks all good.

### CODE REVIEW

**Comment by serano01** [comment]
Looks good.

File: src/object/circleplatform.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
+//  Copyright (C) 2020 Daniel Ward <weluvgoatz@gmail.com>
```
Comment (Zwatotem) [inline src/object/circleplatform.cpp:1]: My editor shows, that this file has mixed indentation (tabs and spaces) please change all to spaces.

File: data/images/engine/editor/objects.stoi [hunk 2]
```diff
@@ -297,6 +297,9 @@
   (object
     (class "skull_tile")
     (icon "images/objects/skull_tile/skull.png"))
+  (object
+    (class "circleplatform")
+    (icon "images/objects/platforms/icebridge1.png"))
```
Comment (Zwatotem) [inline data/images/engine/editor/objects.stoi:302]: Please, could you make dedicated icon (potentialy by modifying bicycle-platform one)?

**Comment by weluvgoatz** [comment]
I updated the pull request by updating with your suggestions and requirements @Zwatotem, but I don't understand your problems with the indentation seen as I cannot see the problems you are talking about.

**Pull request merged**